packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -769,11 +806,28 @@ class _PageViewState extends State<PageView> {
     return null;
   }
 
+  double _getCacheExtent(AxisDirection direction, BoxConstraints constraints) {
+    // Use the actual child count if we can, otherwise use three for leading/active/trailing.
+    final int childCount = widget.childrenDelegate.estimatedChildCount ?? 3;
```

Comment:
This essentially makes the pageview unlazy? You always build all children?

Reply:
In the case where we want to allow implicit scrolling, yes.